File: hospitalManagement/Doctors.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal class Doctors: IBasicActivities<Doctor>
    {
        //field
        private List<Doctor> doctorList;
        private long count;


        // Infor field
        //  Management field
        // Dynamic 
        // Properties
        public long Count { get => count; set => count = value; }
        internal List<Doctor> DoctorList { get => doctorList; set => doctorList = value; }


        // Constructors

        public Doctors()
        {
            doctorList = new List<Doctor>();
            count = 0;
        }
        public Doctors(List<Doctor> doctorList, long count)
        {
            this.doctorList = doctorList;
            this.count = count;
        }

        public void AddItem()
        {
            Console.WriteLine("Add doctor");

            Doctor doctor = new Doctor();
            doctor.Input();
            doctorList.Add(doctor);
            this.Count++;
            Console.WriteLine("Done!");

        }

        public void AddItem(Doctor value)
        {
            Console.WriteLine("Add doctor");

            doctorList.Add(value);
            this.Count++;
            Console.WriteLine("Done!");

        }

        public bool Clear()
        {
            Console.WriteLine("Clear all doctors");

            doctorList.Clear();
            if (doctorList.Count > 0)
            {
                Console.WriteLine("Failure");
                return false;
            }
            this.Count = 0;
            Console.WriteLine("Done!");
            return true;
        }

        public Doctor FindItem(string id)
        {
            Doctor res = null;
            doctorList.ForEach(value =>
            {
                if (value.DoctorId == id)
                {
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found doctor with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public bool RemoveItem(string id)
        {
            Console.WriteLine("Remove the Doctor");

            bool res = false;
            doctorList.ForEach(value =>
            {
                if (value.DoctorId == id)
                {
                    doctorList.Remove(value);
                    res = true;
                    this.Count--;
                }
            });
            if (res == false)
            {
                Console.WriteLine($"Not found doctor with id: {id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }

        public void ShowInformation()
        {
            Console.WriteLine("Show information of all doctors");

            doctorList.ForEach(value => { value.Output(); Console.WriteLine(); });
            Console.WriteLine("Done!");
        }

        public Doctor UpdateItem(string id)
        {
            Console.WriteLine("Update the doctor");

            Doctor res = null;
            doctorList.ForEach(value =>
            {
                if (value.DoctorId == id)
                {
                    value.Input();
                    res = value;
                }
            });
            if (res == null)
            {
                Console.WriteLine($"Not found the doctor with id:{id}");

            }
            else
            {
                Console.WriteLine("Done!");

            }
            return res;
        }
    }
}

File: src/hospitalManagement/Doctors.java
package hospitalManagement;

import java.util.ArrayList;
import java.util.Scanner;

public class Doctors implements IBasicActivities<Doctor>
    {
        //field
        private ArrayList<Doctor> doctorList;
        private long count;



        // Information field
        //  Management field
        // Dynamic 
        static Scanner input = new Scanner(System.in);

        // Properties
        
        public ArrayList<Doctor> getDoctorList() {
            return doctorList;
        }
        public void setDoctorList(ArrayList<Doctor> doctorList) {
            this.doctorList = doctorList;
        }
        public long getCount() {
            return count;
        }
        public void setCount(long count) {
            this.count = count;
        }

        // Constructors

        public Doctors()
        {
            doctorList = new ArrayList<Doctor>();
            count = 0;
        }
        public Doctors(ArrayList<Doctor> doctorList, long count)
        {
            this.doctorList = doctorList;
            this.count = count;
        }

        public void AddItem()
        {
            System.out.println("Add doctor");

            Doctor doctor = new Doctor();
            doctor.Input();
            doctorList.add(doctor);
            this.count++;
            System.out.println("Done!");

        }

        public void AddItem(Doctor value)
        {
            System.out.println("Add doctor");

            doctorList.add(value);
            this.count++;
            System.out.println("Done!");

        }

        public boolean Clear()
        {
            System.out.println("Clear all doctors");

            doctorList.clear();
            
            System.out.println("Done!");
            return true;
        }

        public Doctor FindItem(String id)
        {
            Doctor res = null;
            for (Doctor i : doctorList) {
                if( i.getId() == id)
                {
                    res = i;
                }
            }
            if (res == null)
            {
                System.out.println("Not found doctor with id: " + id);

            }
            else
            {
                System.out.println("Done!");

            }
            return res;
        }

        public boolean RemoveItem(String id)
        {
            System.out.println("Remove the Doctor");

            boolean res = false;
            for (Doctor i : doctorList) {
                if( i.getId() == id)
                {
                    doctorList.remove(i);
                    this.count--;
                }
            }
            if (res == false)
            {
                System.out.println("Not found doctor with id: " + id);

            }
            else
            {
                System.out.println("Done!");

            }
            return res;
        }

        public void ShowInformation()
        {
            System.out.println("Show information of all doctors");

            doctorList.forEach(value -> { value.Output(); System.out.println(); });
            System.out.println("Done!");
        }

        public Doctor UpdateItem(String id)
        {
            System.out.println("Update the doctor");

            Doctor res = null;
            for (Doctor i : doctorList) {
                if( i.getId() == id)
                {
                    i.Input();
                    res = i;
                }
            }
            if (res == null)
            {
                System.out.println("Not found the doctor with id: " + id);

            }
            else
            {
                System.out.println("Done!");

            }
            return res;
        }
    }